src/main/scala/chisel3/aop/Select.scala

```diff
@@ -618,4 +618,50 @@ object Select {
       s"printf when(${preds.map(_.serialize).mkString(" & ")}) on ${getName(clock)}: $pable"
     }
   }
+
+  /** Select.unsafe exposes useful, but unsafe APIs.
+    *
+    * These APIs operate on directly Chisel's internal data structures, which may be mutated as more hardware is
+    * generated. They can only offer a point in time view of the current state. The tradeoff is they do not require the
+    * modules being selected on to have finished construction.
+    */
+  object unsafe {
+
+    /** Selects all instances/modules that are currently directly instantiated within the BaseModule.
+      */
+    def currentInstancesIn(parent: BaseModule): Seq[Instance[BaseModule]] = {
+      parent._ids.collect {
+        case clone: IsClone[_] =>
+          new Instance(Clone(clone)).asInstanceOf[Instance[BaseModule]]
+        case mod: BaseModule =>
+          new Instance(Proto(mod))
+      }.toSeq
```

Comment:
```suggestion
      parent._ids.view.collect {
        case clone: IsClone[_] =>
          new Instance(Clone(clone)).asInstanceOf[Instance[BaseModule]]
        case mod: BaseModule =>
          new Instance(Proto(mod))
      }.toSeq
```
We should get in the habit of using views to avoid extra collection constructions. I know the Chisel codebase is lousy with us not doing this, but I want to encourage it in new code.